File: docs/demos/not_working/Renderer/src/jot/demo/util/GameConstants.java
package jot.demo.util;

import java.util.logging.Logger;
import static java.util.logging.Logger.getLogger;
import static jot.util.CoreOptions.showTextures;
import static jot.util.CoreOptions.showWireframe;
import static jot.util.CoreOptions.useKeyBoard;
import static jot.util.CoreOptions.useMouse;
import static jot.util.CoreOptions.useMouseCursor;
import static jot.util.ExtensionGeometryOptions.useTerrainGenerators;
import static jot.util.ExtensionPhysicsOptions.useCollisionDetection;
import static jot.util.ExtensionPhysicsOptions.useNarrowPhaseCollisionDetection;
import static jot.util.FrameworkOptions.Floor;
import static jot.util.FrameworkOptions.SkyBox;
import static jot.util.FrameworkOptions.SkyDome;
import static jot.util.FrameworkOptions.addSkyBoxCeil;
import static jot.util.FrameworkOptions.addSkyBoxWalls;
import static jot.util.FrameworkOptions.showGeometries;
import static jot.util.FrameworkOptions.showPlanarShadows;
import static jot.util.FrameworkOptions.showShadowMaps;
import static jot.util.FrameworkOptions.showZBuffer;
import static jot.util.FrameworkOptions.useLights;
import static jot.util.FrameworkOptions.useRayTracer;

/**
 * Class to set up the game constants.
 *
 * @author G. Amador & A. Gomes
 */
public class GameConstants {

    static final Logger log = getLogger("GameConstants");

    /**
     * Method to set up the defaults for the game constants options.
     */
    public static void setGameConstants() {
        //Input devices options
        useMouse = true;
        useMouseCursor = true;
        useKeyBoard = true;

        //Geometry options
        showTextures = true;
        showWireframe = false;

        //Geometry generators options
        useTerrainGenerators = false;

        //Physics options
        useCollisionDetection = true;
        useNarrowPhaseCollisionDetection = false;

        //Lights options
        useLights = true;

        //Render options
        useRayTracer = false;

        //Shadows options        
        showPlanarShadows = true;
        showShadowMaps = false;

        //zbuffer options
        showZBuffer = false;

        //Scene options
        showGeometries = true;
        Floor = true;
        SkyDome = false;
        SkyBox = false;
        addSkyBoxCeil = true;
        addSkyBoxWalls = true;

        //TODO: read from file        
    }

    /**
     * Default constructor.
     */
    GameConstants() {
    }
}

File: demos/Physics-Collisions-Shadows/assets/scripts/GameConstants.js
function GameConstant(varName, varValue) {
    print(varName + " = " + varValue);
}

//Input devices options
new GameConstant("useMouse", true);
new GameConstant("useMouseDebug", false);
new GameConstant("useMouseCursor", false);
new GameConstant("useMouseOffScreenPositionShift", true);
new GameConstant("useKeyBoard", true);
new GameConstant("useKeyBoardDebug", false);

//Geometry options
new GameConstant("showTextures", true);
new GameConstant("showWireframe", false);
new GameConstant("useDisplayLists", false);

//Physics options
new GameConstant("useBroadPhaseCollisionDetection", true);
new GameConstant("useNarrowPhaseCollisionDetection", false);
new GameConstant("showPlanarShadows", false);
new GameConstant("showShadowMaps", false);
new GameConstant("useLights", true);

//Z-Buffer
new GameConstant("showZBuffer", false);

//Scene options
new GameConstant("showGeometries", true);
new GameConstant("useUnreachableSky", false);
new GameConstant("useUnreachableFloor", false);
new GameConstant("Floor", true);
new GameConstant("SkyDome", true);
new GameConstant("SkyBox", false);

//Sound options
new GameConstant("useSoundDebug", false);

//GUI options
new GameConstant("showFPS", true);
new GameConstant("showText", true);
new GameConstant("testMode", false);
new GameConstant("useFullScreen", false);